arangod/VocBase/LogicalCollection.cpp

```diff
@@ -253,6 +257,49 @@ LogicalCollection::LogicalCollection(TRI_vocbase_t& vocbase, VPackSlice const& i
   return category;
 }
 
+Result LogicalCollection::updateValidators(VPackSlice validatorArray) {
+  if(validatorArray.isNone()) {
+    return { TRI_ERROR_NO_ERROR };
+  } else if(validatorArray.isArray()) {
+    return { TRI_ERROR_BAD_PARAMETER, "Validators are not given in an array."};
+  }
+
+  try {
+    std::shared_ptr<ValidatorVec> newVec = std::make_shared<ValidatorVec>();
+    for(VPackSlice validatorSlice : VPackArrayIterator(validatorArray)) {
+      if(!validatorSlice.isObject()){
+        return {TRI_ERROR_BAD_PARAMETER, "Validator description is not an object."};
+      }
+
+      std::unique_ptr<ValidatorBase> validator;
+      auto typeSlice = validatorSlice.get(StaticStrings::ValidatorParameterType);
+      if(typeSlice.isNone()) {
+        validator = std::make_unique<ValidatorAQL>(validatorSlice);
+      } else if (typeSlice.isString()) {
+        if (typeSlice.compareString(StaticStrings::ValidatorTypeAQL)) {
+          validator = std::make_unique<ValidatorAQL>(validatorSlice);
+        } else if (typeSlice.compareString(StaticStrings::ValidatorTypeBool)) {
+          validator = std::make_unique<ValidatorBool>(validatorSlice);
+        } else {
+          return {TRI_ERROR_BAD_PARAMETER, "Validator type '" + typeSlice.copyString() + "' is not supported."};
+        }
+      } else {
+        return {TRI_ERROR_BAD_PARAMETER, "Validator type is not a string."};
+      }
+
+      newVec->push_back(std::move(validator));
+    }
+
+    std::atomic_store_explicit(&_validators, newVec, std::memory_order_relaxed);
+  } catch (arangodb::basics::Exception const& ex){
```

Comment:
Can we get away without any try...catch here?
If an exception occurs in this method and escapes from here, some levels above should catch it.

Reply:
Tried it, but using `try / catch` here allows to provide better error codes & messages.